src/test/java/org/jlab/mya/stream/MySamplerStreamTest.java

```diff
@@ -0,0 +1,326 @@
+package org.jlab.mya.stream;
+
+import org.jlab.mya.TimeUtil;
+import org.jlab.mya.event.EventCode;
+import org.jlab.mya.event.FloatEvent;
+import org.jlab.mya.event.IntEvent;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+
+public class MySamplerStreamTest {
+
+    /**
+     * A method for comparing lists of FloatEvents for equality.  FloatEvent does not have an equals() method, so this
+     * is about the best to be done here.
+     *
+     * @param exp    The expected list
+     * @param result The actual list
+     */
+    public static void assertFloatEventListEquals(List<FloatEvent> exp, List<FloatEvent> result) {
+        // Event does not define equals, so this is about as good as it gets.
+        assertEquals("Received a different number of events than expected", exp.size(), result.size());
+        for (int i = 0; i < exp.size(); i++) {
+            assertEquals("Mismatch at event " + i, exp.get(i).toString(), result.get(i).toString());
```

Comment:
This String comparison may fail causing flaky tests as comparing floats is inexact.   The [Assert.assert(expected, actual, delta)](https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertEquals(float,%20float,%20float)) method may be better.  Example:  https://github.com/JeffersonLab/jmyapi/blob/d509e8a9256274f5cd66117cb554d61d42570156/src/test/java/org/jlab/mya/stream/FloatEventBucketTest.java#L103

Reply:
Done.  I also added explicit check for timestamps as well.